src/features/utils/aggregate-base.js

```diff
@@ -47,6 +53,39 @@ export class AggregateBase extends FeatureBase {
     this.drained = true
   }
 
+  /**
+   * Return harvest payload. A "serializer" function can be defined on a derived class to format the payload.
+   * @param {Boolean} shouldRetryOnFail - harvester flag to backup payload for retry later if harvest request fails; this should be moved to harvester logic
+   * @returns final payload, or undefined if there are no pending events
+   */
+  makeHarvestPayload (shouldRetryOnFail = false, opts = {}) {
+    if (this.events.isEmpty(opts)) return
```

Comment:
Not atm but I suppose it could, and you are correct about it only being used by features that use EventAggregator right now. What is your suggestion on being able to pass in the `aggregatorTypes` to this fn shared by all the features?

The complication is that both features using it shares the same instance reference, so you can't "remember" the types in a scoped field. You'd have to make it so when jserror executes `makeHarvestPayload`, its `this.events.get()` implicitly does a `.get(['err','xhr'])` while for metric its `this.events.get()` implicitly does a `.get(['sm','cm'])` (and likewise for the other instance functions). Probably would have to create another layer wherein they hold separate references where the types is set in a closure.

Reply:
Ah, that is tricky.  Initially I had assumed we'd just pass in the `opts` during construction and remember it (so using EventAgg instances) but I guess I don't understand enough to decide if using a singleton or having individual instances of EventAgg would be preferable here.

Reply:
Think I got the gist of it and resolving for now.  Not necessarily a blocker as it seems like a matter of design choice.